### PR TITLE
[INFINITY-1188] fix ParentElement.getStatus

### PR DESCRIPTION
**Pull request by mgummelt**

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java [hunk 1]
```diff
@@ -100,47 +98,48 @@ default Status getStatus() {
...
         // Also note that this function MUST NOT call parent.getStatus() as that creates a circular call.
 
-        final Collection<? extends Element> children = getChildren();
+        final List<C> children = getChildren();
         if (children == null) {
             LOGGER.error("Parent element returned null list of children: {}", getName());
             return Status.ERROR;
         }
 
+        final Collection<? extends Element> candidateChildren =
```
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java]: Parent Status depends on its strategy.  A parent with "PENDING, WAITING" children should be PENDING, whereas a parent with "WAITING, PENDING" children should be WAITING.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java [hunk 2]
```diff
@@ -100,47 +98,48 @@ default Status getStatus() {
...
+                getStrategy().getCandidates(children, Collections.emptyList());
+
         Status result;
-        if (!getErrors().isEmpty()) {
+        if (!getErrors().isEmpty() || anyHaveStatus(Status.ERROR, children)) {
             result = Status.ERROR;
-            LOGGER.debug("({} status={}) Elements contains errors", getName(), result);
-        } else if (CollectionUtils.isEmpty(children)) {
```
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java]: This check is redundant.  It's covered by the following check.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java [hunk 3]
```diff
@@ -100,47 +98,48 @@ default Status getStatus() {
...
-        } else if (allHaveStatus(Status.PENDING, children)) {
-            result = Status.PENDING;
-            LOGGER.debug("({} status={}) All elements have status: {}",
-                    getName(), result, Status.PENDING);
         } else if (anyHaveStatus(Status.PREPARED, children)) {
             result = Status.IN_PROGRESS;
             LOGGER.debug("({} status={}) At least one phase has status: {}",
                     getName(), result, Status.PREPARED);
-        } else if (anyHaveStatus(Status.IN_PROGRESS, children)) {
+        }  else if (anyHaveStatus(Status.WAITING, candidateChildren)) {
```
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java]: A Parent is waiting only if its candidate children are waiting.

**Comment by mgummelt** [comment]
cc @mbdcos for review

**Comment by mbdcos** [comment]
Much better than previous PlanUtils.getStatus()

**Pull request merged**